group/assignment5/sridha2412.py

```diff
@@ -0,0 +1,71 @@
+from typing import List, Tuple
+
+def common_prefix(word1: str, word2: str) -> int:
+    """ A method to find the common prefix in two words and return the end position
+    Args:
+        word1: first word
+        word2: second word
+    Returns:
+        int: position end of common prefix
+    """
+
+    for i in range(0, min(len(word1), len(word2))):
+        if word1[i] != word2[i]:
+            break
```

Comment:
```
    if word1[i] != word2[i]:
        return i

return Null
```

and in the create_dag function:
```
cp = common_prefix(word1, word2)
if cp:
```

This might look better in order not to check the lengths again.

Reply:
I agree with this